saleor/product/search.py

```diff
@@ -103,7 +103,10 @@ def generate_attributes_search_document_value(
             ]
         elif input_type == AttributeInputType.NUMERIC:
             unit = attribute.unit
-            values_list = [value.name + unit for value in values]
+            if unit:
+                values_list = [value.name + unit for value in values]
+            else:
+                values_list = [value.name for value in values]
```

Comment:
We can get rid of the condition part:
```suggestion
            unit = attribute.unit or ""
            values_list = [value.name + unit for value in values]
```